crates/op-rbuilder/src/builders/context.rs

```diff
@@ -344,15 +345,31 @@ impl OpPayloadBuilderCtx {
 
         while let Some(tx) = best_txs.next(()) {
             let interop = tx.interop_deadline();
-            let exclude_reverting_txs = tx.exclude_reverting_txs();
+            let reverted_hashes = tx.reverted_hashes().clone();
+            let conditional = tx.conditional().cloned();
+
             let tx_da_size = tx.estimated_da_size();
             let tx = tx.into_consensus();
             let tx_hash = tx.tx_hash();
 
+            // exclude reverting transaction if:
+            // - the transaction comes from a bundle and the hash **is not** in reverted hashes
+            let exclude_reverting_txs =
+                reverted_hashes.is_some() && !reverted_hashes.unwrap().contains(&tx_hash);
+
             let log_txn = |result: TxnExecutionResult| {
                 info!(target: "payload_builder", tx_hash = ?tx_hash, tx_da_size = ?tx_da_size, exclude_reverting_txs = ?exclude_reverting_txs, result = %result, "Considering transaction");
             };
 
+            if let Some(conditional) = conditional {
+                // TODO: ideally we should get this from the txpool stream
+                if let Some(min_block_number) = conditional.block_number_min {
+                    if min_block_number > self.block_number() {
+                        continue;
+                    }
```

Comment:
If we continue here, does the tx go back to the tx pool and get presented to us again in the future or it gets dropped and we need to keep track of it on the builder side?

Reply:
The transactions are never dropped from the pool unless we call the `remove_transaction` method. But, in this case we would have to mark it as invalid so that we do not consider its childs either like we do with the rest of the code.